### PR TITLE
Some corrections to build unittests on Windows. Try a new option for …

### PR DESCRIPTION
**Pull request by gautierbureau**
…boost build to avoid errors with cmake and MSVC.

### CODE REVIEW

File: dynawo/sources/Common/DYNErrorQueue.h [hunk 1]
```diff
@@ -51,6 +49,13 @@ class DYNErrorQueue  : private boost::noncopyable{
    */
   void flush();
 
+  /**
+   * @brief Get the maximum number of errors displayed
+   *
+   * @return maximum number of errors displayed
+   */
+  size_t getMaxDisplayedError();
```
Comment (rosiereflo) [inline dynawo/sources/Common/DYNErrorQueue.h]: const

File: dynawo/sources/Common/gtest_dynawo.h [hunk 2]
```diff
@@ -116,7 +119,34 @@ inline std::string key2Str(const int key) {
     oss << "  Actual: it throws an exception of unknown type";                                   \
     GTEST_FATAL_FAILURE_(oss.str().c_str());                                                     \
   }
-
+#else
+#define ASSERT_THROW_DYNAWO(F, ERROR_TYPE, ERROR_KEY)                                            \
+  try                                                                                            \
```
Comment (rosiereflo) [inline dynawo/sources/Common/gtest_dynawo.h]: Please limit the ifdef to the modified code

File: dynawo/sources/Common/test/Test.cpp [hunk 3]
```diff
@@ -334,7 +334,11 @@ TEST(CommonTest, testSparseMatrix) {
   SparseMatrix smj;
   smj.init(3, 3);
   smj.changeCol();
+#ifndef _MSC_VER
   smj.addTerm(1, 1./0.);
+#else
+  smj.addTerm(1, 1.);
```
Comment (rosiereflo) [inline dynawo/sources/Common/test/Test.cpp]: We are testing the presence of a Nan in the matrix so the test is not equivalent

File: dynawo/sources/Modeler/Common/test/Test.cpp [hunk 4]
```diff
@@ -688,8 +688,9 @@ TEST(ModelerCommonTest, VariableAlias) {
   ASSERT_EQ(variableAlias3->getNegated(), true);
   ASSERT_EQ(variableAlias3->getIndex(), varIndex);
 
+#ifndef _MSC_VER
   EXPECT_ASSERT_DYNAWO(VariableAliasFactory::create(aliasName3, dynamic_pointer_cast<VariableNative> (variableAlias2), DISCRETE));
```
Comment (rosiereflo) [inline dynawo/sources/Modeler/Common/test/Test.cpp]: Why is this line not working on Windows?

File: CPPLINT.cfg [hunk 5]
```diff
@@ -13,3 +13,4 @@ filter=-build/header_guard
 filter=-build/c++tr1
 filter=-runtime/references
 filter=-runtime/int
+filter=-build/endif_comment
```
[the remark this answers is not in the document]
Comment (gautierbureau) [reply CPPLINT.cfg]: done

File: README.md [hunk 6]
```diff
@@ -202,6 +202,8 @@ Open `x64 Native Tools Command Prompt for VS2019` and run the following commands
 ``` batch
 > git config --global core.eol lf
 > git config --global core.autocrlf input
+> set OPENMODELICAHOME=
```
[the remark this answers is not in the document]
Comment (gautierbureau) [reply README.md]: No it is Windows command not Linux

File: dynawo/sources/Common/DYNErrorQueue.h (hunk 1 above)
Comment (gautierbureau) [reply dynawo/sources/Common/DYNErrorQueue.h]: done

File: dynawo/sources/Common/gtest_dynawo.h (hunk 2 above)
Comment (gautierbureau) [reply dynawo/sources/Common/gtest_dynawo.h]: It is not possible to use ifdef in a macro

File: dynawo/sources/Common/test/Test.cpp (hunk 3 above)
Comment (gautierbureau) [reply dynawo/sources/Common/test/Test.cpp]: I know but MSVC raise an error when it catches / 0. and I don't know how to ignore it...

File: dynawo/sources/Modeler/Common/test/Test.cpp (hunk 4 above)
Comment (gautierbureau) [reply dynawo/sources/Modeler/Common/test/Test.cpp]: it uses ::testing::KilledBySignal(SIGABRT) that is not available on Windows

File: CPPLINT.cfg (hunk 5 above)
Comment (rosiereflo) [inline CPPLINT.cfg]: Why do  we need to disable this rule?

File: dynawo/sources/Common/test/Test.cpp (hunk 3 above)
Comment (rosiereflo) [reply dynawo/sources/Common/test/Test.cpp]: we can try to use NAN from math.h

File: CPPLINT.cfg (hunk 5 above)
Comment (gautierbureau) [reply CPPLINT.cfg]: cpplint rule : Uncommented text after #endif is non-standard. In the case of ifdef MSVC that closes 2 lines below i find this a bit extreme... I can put it back if needed

File: dynawo/sources/Common/test/Test.cpp (hunk 3 above)
Comment (gautierbureau) [reply dynawo/sources/Common/test/Test.cpp]: It was a test for inf and not nan. I just defined a variable with zero and the compiler does not complain this time

File: CPPLINT.cfg (hunk 5 above)
Comment (rosiereflo) [reply CPPLINT.cfg]: Always better to keep coding style guards

**Pull request merged**